.ci/diff-report.sh

```diff
@@ -3,6 +3,36 @@ set -e
 
 source ./.ci/util.sh
 
+# Validates the URL, checks if it is a raw link, and exits if it is not. We
+# expect users to provide a raw gist or GitHub link.
+validate_url () {
+  URL=$1
+
+  VALID_URL_REGEX='^(https?|ftp|file)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]'
+  VALID_URL_REGEX+='\.[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$'
+
+  if [[ -z "$URL" ]]; then
+    # It is ok for a URL to be empty, we use this later in the workflow
+    # to determine what "mode" we should generate report with.
+    echo "URL is empty."
+  elif [[ "$URL" == *: ]]; then
+    # We were not able to match in previous 'sed' command, so user has
+    # incorrectly supplied parameters in PR description.
+    echo "Parameter '${URL}' is incorrectly formatted, please use the following format:"
+    echo "'Parameter name: https://gist.githubusercontent.com/username/gist_id/raw/file'"
+    echo "Parameter name and URL must be separated by a colon and single space only."
+    exit 1
+  elif [[ "$URL" != *"raw"* ]]; then
+    echo "URL '${URL}' must be a direct raw link to a gist or GitHub file."
+    exit 1
+  elif ! [[ "$URL" =~ $VALID_URL_REGEX ]]; then
+    echo "URL '${URL}' is invalid."
```

Comment:
Not matching validation regexp.
Make sense to print regexp here.
Regexp can be wrong also.